internal/util/rocksmq/server/rocksmq/rocksmq_retention.go

```diff
@@ -76,10 +76,10 @@ func prefixLoad(db *gorocksdb.DB, prefix string) ([]string, []string, error) {
 	for ; iter.Valid(); iter.Next() {
 		key := iter.Key()
 		value := iter.Value()
-		defer key.Free()
-		defer value.Free()
 		keys = append(keys, string(key.Data()))
+		key.Free()
 		values = append(values, string(value.Data()))
```

Comment:
Is there also a suspicion of accessing dangling pointers here?
refer to https://github.com/milvus-io/milvus/issues/7300

Reply:
NO!

Reply:
why? Is string conversion guaranteed to copy data?